### ISSUE TITLE
Getting "Refused to evaluate a string as JavaScript..." when trying to save a .less file.

### ISSUE THREAD

**Issue by RomansBermans** [Metric/Less-Compile#2]
Refused to evaluate a string as JavaScript because 'unsafe-eval' is not an allowed source of script in the following Content Security Policy directive: "script-src 'self'".

**Comment by morcegon** [comment]
How can I install loophole to avoid this error? Thanks.

**Comment by Metric** [comment]
What version of Atom? I haven't updated the plugin for the latest version
of it yet. I am still using 0.177.0, because the guys creating Atom can't
seem to make up their minds for their underlying plugin apis. Almost every
other update they change something in the apis it seems like.

I implemented the loophole into the actual implementation of the plugin,
but if it isn't working in the latest build. Then that means they changed
something to prevent the previous loophole method.

**Comment by morcegon** [comment]
I'm use the latest version. The error was showed when I tried to compile a lesshat function.

**Comment by Metric** [comment]
Could you link me to an example less file of what you are trying to do? I
would like to test it with the version of Atom I have, to see if the older
version throws the same error.

**Comment by Metric** [comment]
I can confirm that lesshat.less does compile on the older version of Atom 0.177.0. So, they did indeed change something on the new version of Atom.

**Comment by zenyr** [comment]
I monkey-patched `var Function = require('loophole').Function;` at `\.atom\packages\less-autocompile\node_modules\less\lib\less\tree\js-eval-node.js` to fix this issue temporarily. 
*yeah that's right, I modified `less` module which is stupid to begin with, but hey it works and won't break anything else. *

A proper fix would be appreciated.

**Comment by Metric** [comment]
I still would like to know what lesshat function is causing this.
Everything works fine on the latest version of atom on Mac for me. I have
tried multiple lesshat functions that use JavaScript and they all work with
no error.

Are you running the latest version of Windows Atom, Mac Atom or Linux Atom?

**Comment by zenyr** [comment]
In my case I was on atom 1.0.5 @ Windows. And I was not using lesshat, but that is not the problem I think. The code was this javascript eval line for splitting an image URL to insert @2x string between them:
````less
@filename  : ~`/(.*)\.(jpg|jpeg|png|gif)/.exec(@{image})[1]`;
@extension : ~`/(.*)\.(jpg|jpeg|png|gif)/.exec(@{image})[2]`;
````
:)

**Comment by Metric** [comment]
Okay, so it appears to be an issue with Atom 1.0.5. I just updated to it on Mac, and boom lesshat now throws the same error when using .box-sizing again. Atom 1.0.4 and lower seems to work just fine.

**Comment by Metric** [comment]
I have opened an issue with Atom to see what is up with this. As it appears to be 1.0.5 causing this issue, while anything lower still worked as expected.

**Comment by morcegon** [comment]
Any update about this issue? Thanks.

**Comment by Metric** [comment]
Oops, sorry. The atom team and I came to the conclusion that it must be another plugin installed conflicting with the loop hole method on save for you guys that are experiencing this.

Everything is working fine here on Windows and Mac with the latest versions of atom. I have not made any changes to the code. I am only testing with this plugin and uglify plugin I created installed. The only other plugins installed are the default that come with atom. This is why the atom team and I believe it is another plugin causing the loophole fix I use to be overwritten before the less processor can fully complete.
